drivers/mipi_dbi/mipi_dbi_nxp_lcdic.c

```diff
@@ -413,6 +426,24 @@ static int mipi_dbi_lcdic_write_display(const struct device *dev,
 		goto out;
 	}
 
+	if (dev_data->new_frame) {
+		switch (dev_data->te_edge) {
+		case MIPI_DBI_TE_RISING_EDGE:
+			te_sync = LCDIC_TE_RISING_EDGE;
+			break;
+		case MIPI_DBI_TE_FALLING_EDGE:
+			te_sync = LCDIC_TE_FALLING_EDGE;
+			break;
+		default:
+			te_sync = LCDIC_TE_NO_SYNC;
+			break;
+		}
+		dev_data->new_frame = false;
+	} else if (!desc->frame_incomplete) {
+		/* Next frame will be a new one */
+		dev_data->new_frame = true;
+	}
```

Comment:
I'm a little confused here; the way this `if` statement is written, if every single frame is `frame_incomplete = false`, then the `te_sync` will only be active every other `write` call. Is that intended?

I feel like the `else` should be removed, and the `if (!desc->frame_incomplete)` should be executed always.

Feel free to correct my thoughts if I'm mistaken.

Reply:
No, I think you are right here. We always want the next call after one with `frame_incomplete = false` to trigger a TE wait. I'll update this.

Reply:
Updated